socs/agent/class_ps3000a.py

```diff
@@ -0,0 +1,519 @@
+import ctypes
+from picosdk.ps3000a import ps3000a as ps
+from picosdk.functions import adc2mV, assert_pico_ok, splitMSODataFast
+import time
+import datetime
+import numpy as np
+
+class ps3000a():
+    def __init__(self, sizeofbuffer, samplerate):
+        self.status = {}
+        self.chandle = ctypes.c_int16()
+        self.interval = int(1./samplerate*1e9) # unit ns
+        
+        # Size of capture
+        self.sizeOfOneBuffer = sizeofbuffer
+        self.numBuffersToCapture = 1
+        self.totalSamples = self.sizeOfOneBuffer * self.numBuffersToCapture
+        
+        print('sample points: {}, sampling rate: {} (MHz), length: {} (sec)'.format(self.totalSamples, samplerate*1e-6, self.totalSamples*self.interval*1e-9))
+        
+        # Opens the device/s
+        self.status["openunit"] = ps.ps3000aOpenUnit(ctypes.byref(self.chandle), None)

```

Comment:
Yes, in this line communication with picoscope is initialted.
Since we don't do streaming with this agent, every time when we run the `run_signle`, this agent initiate the communication with picoscope and stop the communication.
Currently this is working. Should I fix to initiate the communication only once?

Reply:
I see. It's probably alright as is. How often do we expect to call the `run_single` Task?